src/components/Canvas.tsx

```diff
@@ -1,23 +1,30 @@
 import React from 'react';
 import Node from './Node';
 import Edge from './Edge';
-import { LabeledNode, LabeledEdge } from '../logic/commonInterfaces';
+import { LabeledNode, LabeledEdge, Frame } from '../logic/commonInterfaces';
 
 
 
-function Canvas({nodes, edges}
+function Canvas({nodes, edges, frame}
   : {
     nodes: Array<LabeledNode>,
     edges: Array<LabeledEdge>,
+    frame: Frame,
   })
 {
   let edgeComponents, nodeComponents;
   if (nodes.length) {
     nodeComponents = nodes.map( node =>
       <Node x={node.x} y={node.y} label={node.label} key={node.id} />,
     );
-    edgeComponents = edges.map( edge =>
-      <Edge from={edge.from} to={edge.to} key={edge.id} />,
+    edgeComponents = edges.map( (edge, id) =>
+      <Edge
+        from={edge.from}
+        to={edge.to}
+        key={id}
+        label={edge.label}
+        isActive={frame.activeEdges.includes(id)}
+        isInactive={frame.inactiveEdges.includes(id)} />,
```

Comment:
It's weird passing `isActive` and `isInactive` simultaneously. By "inactive" u mean the "blue" color? If yes, then maybe let's find a better name 🙂